### PR TITLE
Implement basics of Tags and HeldItemChangeClientbound packets

### PR DESCRIPTION
**Pull request by Schuwi**
This PR aims to implement the basics of the Tags packet - right now primarily used to make sure lava is rendered correctly in the client - and the HeldItemChangeClientbound packet - sent at server join to set the players selected hotbar slot to the one they had selected on their last login.
Tags are a new feature of 1.13 that specify common traits of block types/item types/fluid types and could be sourced from the vanilla Data Generators. For now I only hardcoded the fluid tags for lava and water to fix the vanilla client rendering lava with a water texture without a tags packet and instead intend to wait on a decision in #267 for a full-fledged implementation of tags.

### CODE REVIEW

File: core/network/src/packets.rs [hunk 1]
```diff
@@ -2406,3 +2414,65 @@ pub struct EntityTeleport {
...
+            &mut self.fluid_tags,
+        ]
+        .iter_mut()
+        {
+            let length = buf.try_get_var_int()?;
+            for _tag in 0..length {
+                let identifier = buf.try_get_string()?;
+                let inner_length = buf.try_get_var_int()?;
+
+                let mut entries = Vec::with_capacity(inner_length as usize);
```
Comment (caelunshun) [inline core/network/src/packets.rs:2439]: Avoid calling `with_capacity` with untrusted inputs—it makes DoS attacks trivial :) A check here to ensure the length isn't too large would be good.

Comment (Schuwi) [reply core/network/src/packets.rs:2439]: Thanks, good tip! Though I'll just remove the `read_from` implementation completely, it's never needed anyway and the new packet lib is coming up

Comment (Schuwi) [reply core/network/src/packets.rs:2439]: Never needed as in this is actually a Server -> Client only packet

**Comment by CDFN** [comment]
Apparently it closes #157 - can't reproduce after changes.

**Pull request merged**